openeo_driver/views.py

```diff
@@ -1316,7 +1318,7 @@ def _download_job_result(
             _log.error(f"Unsupported job result: {result!r}")
             raise InternalException("Unsupported job result")
 
-    def _stream_from_s3(s3_url, result, bytes_range: Optional[str]):
+    def _stream_from_s3(s3_url, filename, mimetype: Optional[str], bytes_range: Optional[str]):
```

Comment:
API design tip: when number of arguments grows, positional argument handling gets more error-prone. You can [enforce keyword argument usage](https://peps.python.org/pep-3102/) with (for example):
```suggestion
    def _stream_from_s3(s3_url, *, filename, mimetype: Optional[str], bytes_range: Optional[str]):
```